crates/spl_network/src/endpoint.rs

```diff
@@ -116,6 +116,25 @@ impl Endpoint {
                     warn!("Failed to send UDP datagram via SPL socket: {error:?}")
                 }
             }
+            OutgoingMessage::VisualReferee(message) => {
```

Comment:
Idea for deduplication with the `OutgoingMessage::GameController`: You could keep the two match-case-arms and in each convert the message very early into `Vec<u8>`.  The rest could be refactored into a shared function to reduce code duplication.